app/src/main/java/com/android/wildex/ui/post/PostDetailsScreen.kt

```diff
@@ -95,18 +141,245 @@ fun PostDetailsScreen(
         IconButton(onClick = onGoBack) {
           Icon(
               imageVector = Icons.AutoMirrored.Filled.ArrowBack,
-              contentDescription = "Back to Homepage")
+              contentDescription = "Back to Homepage",
+              tint = MaterialTheme.colorScheme.primary)
         }
       })
 }
 
-@Preview
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun ProfileScreenPreview() {
-  WildexTheme { Surface(modifier = Modifier.fillMaxSize()) { PostDetailsScreen("fakePostId") } }
+fun PostInfoBar(
+    modifier: Modifier = Modifier,
+    authorId: Id = "",
+    authorProfilePictureURL: URL = "",
+    authorUserName: String = "",
+    animalId: Id = "",
+    date: String = "",
+    location: String = "",
+    likedByCurrentUser: Boolean = false,
+    likesCount: Int = 0,
+    onProfile: (Id) -> Unit = {},
+    postDetailsScreenViewModel: PostDetailsScreenViewModel
+) {
+  Row(
+      modifier = modifier.fillMaxWidth().padding(horizontal = 8.dp, vertical = 8.dp),
+      horizontalArrangement = Arrangement.SpaceBetween,
+      verticalAlignment = Alignment.CenterVertically) {
+        ClickableProfilePicture(
+            modifier = Modifier.size(64.dp).testTag(ProfileScreenTestTags.POSTER_PROFILE_PICTURE),
+            profilePictureURL = authorProfilePictureURL,
+            profileId = authorId,
+            onProfile = onProfile)
+
+        Column(
+            modifier = Modifier.weight(2f).padding(horizontal = 8.dp),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              Text(
+                  text = "$authorUserName saw a",
+              )
+              Text(
+                  text = animalId,
+                  color = MaterialTheme.colorScheme.primary,
+              )
+              Text(
+                  text = date,
+                  color = MaterialTheme.colorScheme.tertiary,
+              )
+            }
+
+        Column(
+            modifier = Modifier.weight(1f).padding(horizontal = 8.dp),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              Icon(
+                  imageVector = Icons.Default.LocationOn,
+                  contentDescription = "Location",
+                  modifier = Modifier.size(40.dp),
+                  tint = MaterialTheme.colorScheme.tertiary)
+
+              Text(
+                  text = location,
+                  color = MaterialTheme.colorScheme.tertiary,
+              )
+            }
+
+        Column(
+            modifier = Modifier.weight(1f).padding(horizontal = 8.dp),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              IconButton(
+                  onClick = {
+                    if (likedByCurrentUser) postDetailsScreenViewModel.addLike()
+                    else postDetailsScreenViewModel.removeLike()
```

Comment:
I think the condition needs to be negated.
```suggestion
                    if (!likedByCurrentUser) postDetailsScreenViewModel.addLike()
                    else postDetailsScreenViewModel.removeLike()
```

Reply:
Yes, good catch